app/models/papi_client.rb

```diff
@@ -0,0 +1,307 @@
+require 'google/apis/genomics_v2alpha1'
+
+##
+# PapiClient: a lightweight wrapper around the Google Cloud Genomics V2 Alpha API for submitting/reporting
+# scp-ingest-service jobs to ingest user-uploaded data to Firestore
+#
+# requires: googleauth, google-api-client, FireCloudClient class (for bucket access)
+#
+# Author::  Jon Bistline  (mailto:bistline@broadinstitute.org)
+
+class PapiClient < Struct.new(:project, :service_account_credentials, :service)
+
+  extend ErrorTracker
+
+  # Service account JSON credentials
+  SERVICE_ACCOUNT_KEY = !ENV['SERVICE_ACCOUNT_KEY'].blank? ? File.absolute_path(ENV['SERVICE_ACCOUNT_KEY']) : ''
+  # Google authentication scopes necessary for running pipelines
+  GOOGLE_SCOPES = %w(https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/datastore)
+  # GCP Compute project to run pipelines in
+  COMPUTE_PROJECT = ENV['GOOGLE_CLOUD_PROJECT'].blank? ? '' : ENV['GOOGLE_CLOUD_PROJECT']
+  # Docker image in GCP project to pull for running ingest jobs
+  INGEST_DOCKER_IMAGE = 'gcr.io/broad-singlecellportal-staging/ingest-pipeline:0.3.1_2959221'
```

Comment:
Is it critical to make `INGEST_DOCKER_IMAGE` configurable before launch?

If so, what kind of interface would be best to update values like "gcr.io/broad-singlecellportal-staging/ingest-pipeline:0.3.1_2959221" -- an environment variable set via CLI argument, an admin GUI, or perhaps something else?

Reply:
I agree that this should be configurable eventually.  There are a variety of ways that we could achieve this.  I think the most robust would be to use the `AdminConfiguration` class and create a new entry type for "Ingest Pipeline Docker Image".  For now, the constant can suffice.

I've created [SCP-1872](https://broadworkbench.atlassian.net/browse/SCP-1872) to account for this.